common/display/displayplanestate.cpp

```diff
@@ -202,7 +202,15 @@ void DisplayPlaneState::ResetLayers(const std::vector<OverlayLayer> &layers,
 
 void DisplayPlaneState::UpdateDisplayFrame(const HwcRect<int> &display_frame) {
   HwcRect<int> &target_display_frame = private_data_->display_frame_;
-  CalculateRect(display_frame, target_display_frame);
+  if (!IsVideoPlane())
+    CalculateRect(display_frame, target_display_frame);
+  else {
+    target_display_frame.bounds[0] = display_frame.bounds[0];
```

Comment:
Hmm, instead of IsVideoPlane, the check should be private_data_->source_layers_.size() == 1, you can just do target_display_frame = display_frame;